build/commands/scripts/sync.js

```diff
@@ -41,40 +50,209 @@ const maybeInstallDepotTools = (options = config.defaultOptions) => {
+                  JSON.stringify(latestSuccessfulSyncInfo, null, 4)}`)
+  } else {
+    console.log(
+        chalk.green.bold(`Chromium repo does not need sync as it is already ${
+            chalk.italic(
+                chromiumRef)} at commit ${targetSHA || '[missing]'}.`))
+  }
+  return needsUpdate
+}
+
+function syncChromium(program) {
+  const requiredChromiumRef = config.getProjectRef('chrome')
+  let args = [
+    'sync', '--nohooks', '--reset', '--revision',
+    'src@' + requiredChromiumRef, '--with_tags',
+    '--with_branch_heads', '--upstream'
+  ];
+
+  const syncWithForce = program.init || program.force
+  if (syncWithForce) {
+    args.push('--force')
+  }
+
+  if (program.sync_chromium_and_delete_unused_deps) {
+    if (util.isGitExclusionExists(config.srcDir, 'brave/')) {
+      args.push('-D')
+    } else {
+      Log.warn(
+          '--sync_chromium_and_delete_unused_deps was specified but cannot ' +
+          'be used to remove old Chromium deps as sync has not yet added the ' +
+          'exclusion for the src/brave/ directory, likely because sync has ' +
+          'not previously successfully run before.')
+    }
+  }
+
+  const latestSuccessfulSyncFilePath =
+      path.join(config.rootDir, '.brave_latest_successful_sync.json')
+  const latestSuccessfulSyncInfo = util.readJSON(latestSuccessfulSyncFilePath)
+  const expectedSuccessfulSyncInfo = {
+    chromiumRef: requiredChromiumRef,
+    gClientTimestamp: fs.statSync(config.gClientFile).mtimeMs.toString(),
+  }
+
+  const chromiumNeedsUpdate =
+      shouldUpdateChromium(latestSuccessfulSyncInfo, expectedSuccessfulSyncInfo)
+  const shouldSyncChromium =
+      chromiumNeedsUpdate || syncWithForce || program.sync_chromium
+  if (!shouldSyncChromium) {
+    return false
+  }
+
+  if (program.sync_chromium !== undefined) {
+    if (!program.sync_chromium) {
+      Log.warn(
+          'Chromium needed sync but received the flag to skip performing the ' +
+          'update. Working directory may not compile correctly.')
+      return false
+    } else if (!chromiumNeedsUpdate && !syncWithForce) {
+      Log.warn(
+          'Chromium doesn\'t need sync but received the flag to do it anyway.')
+    }
+  }
+
+  util.runGClient(args, {cwd: config.rootDir}, config.gClientFile)
+  util.addGitExclusion(config.srcDir, 'brave/')
+  util.writeJSON(latestSuccessfulSyncFilePath, expectedSuccessfulSyncInfo)
+
+  const postSyncChromiumRef = util.getGitReadableLocalRef(config.srcDir)
+  Log.status(`Chromium is now at ${postSyncChromiumRef || '[unknown]'}`)
+  return true
+}
+
+function syncBrave(program) {
+  let args = ['sync', '--nohooks']
+  const syncWithForce = program.init || program.force
+  if (syncWithForce) {
+    args.push('--force')
+  }
+
+  util.runGClient(
+      args, {cwd: config.braveCoreDir},
+      path.join(config.braveCoreDir, '.brave_gclient'))
+}
+
+function gclientRunhooks() {
+  util.runGClient(['runhooks'], {cwd: config.rootDir}, config.gClientFile)
```

Comment:
does this still run hooks in brave/DEPS?

Reply:
I guess it does based on https://github.com/brave/brave-core/pull/14735/files#diff-3197dec97385e41ae374d64a58a83f44937dcdc83c4d30952c9bf56d45cb9d26R242